### ISSUE TITLE
Warped AR for direct vpy input produced by DGSource

### ISSUE THREAD

**Issue by JJKylee** [DJATOM/x265-aMod#8]
Direct `vpy` input by DGSource, seems to warp the aspect ratio of the output hevc file, although the input resolution is _correctly_ provided by the `vpy` file according to the log. (But the **SAR is reversed**.)

```
x265 3.4+65-aMod(GCC10.2.1) DJATOM

D:\Utilities\StaxRip\Apps\Encoders\x265\x265.exe --crf 27 --output-depth 10 --output D:\Downloads\test_1audio_temp\test_1audio_DGSource_x265-direct-input_out.hevc D:\Downloads\test_1audio_temp\test_1audio_DGSource_x265-direct-input.vpy

vpy  [info]: 1920x1080 fps 24000/1001 i420p8 sar 9:16 frames 0 - 2478 of 2479
vpy  [info]: using 4 parallel requests
raw  [info]: output file: D:\Downloads\test_1audio_temp\test_1audio_DGSource_x265-direct-input_out.hevc
x265 [info]: HEVC encoder version 3.4+65-2ce978dd9 [DJATOM's Mod]
...
```

The `vpy` file itself is very simple.

```
import os, sys
import vapoursynth as vs
core = vs.get_core()

sys.path.append(r"D:\Utilities\StaxRip\Apps\Plugins\VS\Scripts")
core.std.LoadPlugin(r"D:\Utilities\StaxRip\Settings\Plugins\Dual\DGDecNV\DGDecodeNV.dll")
clip = core.dgdecodenv.DGSource(r"D:\Downloads\test_1audio_temp\test_1audio.dgi", deinterlace=0, fieldop=0)
clip.set_output()
```

See the log here: 
[test_1audio_DGSource_x265-direct-input_staxrip.log](https://github.com/DJATOM/x265-aMod/files/5906901/test_1audio_DGSource_x265-direct-input_staxrip.log)

OTOH, if I pipe the `vpy` via `vspipe`, this issue does **not** occur.

Here's the log: 
[test_1audio_DGSource_vspipe_staxrip.log](https://github.com/DJATOM/x265-aMod/files/5906905/test_1audio_DGSource_vspipe_staxrip.log)

Please see [these results](https://drive.google.com/drive/folders/1R1jBzfCAn6d61kO5vDSDCQtk-pNJA7mU?usp=sharing) on my Google Drive.

For comparison, I switched the source filter to [StvG's latest ffms2 build](https://forum.doom9.org/showthread.php?p=1928993#post1928993), but with this source filter, even **direct** `vpy` input does **not** raise this problem.

This time the SAR is not warped.

```
vpy  [info]: 1920x1080 fps 24000/1001 i420p8 sar 1:1 frames 0 - 2478 of 2479
```

Can you please take a look into this?

**Comment by DJATOM** [comment]
vspipe [simply puts 0:0](https://github.com/vapoursynth/vapoursynth/blob/master/src/vspipe/vspipe.cpp#L346), so filter's reported AR is just being ignored. To address this I'll introduce option to conform filter's sar if needed. This will exhibit default vspipe behaviour and allow user to follow filter's reported AR if needed.

**Comment by JJKylee** [comment]
Thanks a million, Master. 🙇‍♂️😄

**Comment by DJATOM** [comment]
[x265-x64-test.zip](https://github.com/DJATOM/x265-aMod/files/5909099/x265-x64-test.zip)
Could you please test it? Now SAR assignment explicitly disabled (mimic vspipe) and can be enabled with `--reader-options use-script-sar=1`
As for DG stuff, looks like Don setting it reversed. `vid2 = core.std.SetFrameProps(vid2, __SARNum=16, __SARDen=9)` results in valid (`1920x1080 fps 24000/1001 i420p8 sar 16:9 frames 0 - 2149 of 2150`) SAR reported from reader.

**Comment by JJKylee** [comment]
> [x265-x64-test.zip](https://github.com/DJATOM/x265-aMod/files/5909099/x265-x64-test.zip)
> Could you please test it? Now SAR assignment explicitly disabled (mimic vspipe) and can be enabled with `--reader-options use-script-sar=1`

This build works exactly as you told me. Thanks a million. 👍

> As for DG stuff, looks like Don setting it reversed. `vid2 = core.std.SetFrameProps(vid2, __SARNum=16, __SARDen=9)` results in valid (`1920x1080 fps 24000/1001 i420p8 sar 16:9 frames 0 - 2149 of 2150`) SAR reported from reader.

What is confusing is that **StaxRip does not show a distorted AR in Preview**. Maybe there's some hidden feature in it. I hope @stax76 and @Dendraspis take a look at it. 😄

**Comment by DJATOM** [comment]
Then it's Don's fault, hmm... He should stuff it into _DARxxx frame props or so.

**Comment by DJATOM** [comment]
Released [3.4+70](https://github.com/DJATOM/x265-aMod/releases/tag/3.4%2B70)
